packages/@lwc/engine-core/src/framework/modules/attr-unless-prop.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import {
+    htmlAttributeToProperty,
+    isNull,
+    isUndefined,
+    StringCharCodeAt,
+    XML_NAMESPACE,
+    XLINK_NAMESPACE,
+} from '@lwc/shared';
+
+import { unlockAttribute, lockAttribute } from '../attributes';
+import { RendererAPI } from '../renderer';
+import { EmptyObject } from '../utils';
+import { VBaseElement } from '../vnodes';
+
+const ColonCharCode = 58;
+
+export function patchAttrUnlessProp(
+    oldVnode: VBaseElement | null,
+    vnode: VBaseElement,
+    renderer: RendererAPI
+) {
+    const {
+        data: { attrs },
+        elm,
+    } = vnode;
+    if (isUndefined(elm) || isUndefined(attrs)) {
+        return;
+    }
+
+    const { removeAttribute, setAttribute } = renderer;
+    const oldAttrs = isNull(oldVnode) ? EmptyObject : oldVnode.data.attrs;
+
+    for (const name in attrs) {
+        const cur = attrs[name];
+        const old = oldAttrs[name];
+
+        const propName = htmlAttributeToProperty(name);
+        if (propName in elm) {
+            (elm as any)[propName] = cur;
+        } else if (old !== cur) {
+            unlockAttribute(elm, name);
```

Comment:
I don't think we need lock, and unlock, because the only case in which this matters is if the component is LWC and it is reacting to those changes... but the problem here is that if that was an lwc component, we would be setting the prop instead, which does the locking/unlocking under the hood. I think it is fine to ignore this part in this module.